src/index.ts

```diff
@@ -263,16 +263,22 @@ export default class Wallet {
    * @param icapDirect setting this to `true` will generate an address suitable for the `ICAP Direct mode`
    */
   public static generate(icapDirect: boolean = false): Wallet {
-    if (icapDirect) {
-      const max = new BN('088f924eeceeda7fe92e1f5b0fffffffffffffff', 16)
-      while (true) {
-        const privateKey = randomBytes(32) as Buffer
-        if (new BN(privateToAddress(privateKey)).lte(max)) {
-          return new Wallet(privateKey)
+    // Note that randomBytes(32) can not pass isValidPrivate with very low probability
+    // see more https://raw.githubusercontent.com/paulmillr/noble-secp256k1/master/test/vectors/privates.json
+    while (true) {
+      try {
+        if (icapDirect) {
+          const max = new BN('088f924eeceeda7fe92e1f5b0fffffffffffffff', 16)
+          const privateKey = randomBytes(32) as Buffer
+          if (new BN(privateToAddress(privateKey)).lte(max)) {
+            return new Wallet(privateKey)
+          }
+        } else {
+          return new Wallet(randomBytes(32))
         }
+      } catch (ignoreError) {
+        // Do not crash
```

Comment:
Thanks for the explanation.

Can we update here that only this specific error is ignored and all other errors rethrown? Otherwise all other errors will get hidden with this implementation.

Also: can you add a test case which would fail before the change and pass afterwards?

Reply:
I made better implementation, but it's almost impossible to mock randomBytes. Because randombytes module uses 
`module.exports = require('crypto').randomBytes`
this implementation. So mocking crypto.randomBytes will have no effect, mocking result of require is impossible (or I don't know how to do that), and mocking whole require is kinda overkill